utils/rancher/rancherutil.py

```diff
@@ -0,0 +1,69 @@
+import logging
+import requests
+from rancher_metadata import MetadataAPI
+
+from utils.singleton import Singleton
+
+log = logging.getLogger(__name__)
+
+
+class RancherUtil:
+    __metaclass__ = Singleton
+
+    METADATA_URL_AUTHORITY = "http://rancher-metadata/"
+    METADATA_VERSION = "2016-07-29"
+    METADATA_URL = "{}/{}".format(METADATA_URL_AUTHORITY, METADATA_VERSION)
+
+    CONTAINER_NAME_LABEL = "io.rancher.container.name"
+    CONTAINER_IP_LABEL = "io.rancher.container.ip"
+    STACK_NAME_LABEL = "io.rancher.stack.name"
+    SERVICE_NAME_LABEL = "io.rancher.stack_service.name"
+
+    HOST_AGENT_IMAGE_LABEL = "io.rancher.host.agent_image"
+    HOST_DOCKER_VERSION_LABEL = "io.rancher.host.docker_version"
+    HOST_LINUX_KERNEL_VERSION_LABEL = "io.rancher.host.linux_kernel_version"
+
+    _is_rancher = None
+
+    def __init__(self):
+        self.api = MetadataAPI(api_url=RancherUtil.METADATA_URL)
+
+    @staticmethod
+    def is_rancher():
+        if RancherUtil._is_rancher is None:
+            try:
+                response = requests.get(url=RancherUtil.METADATA_URL)
```

Comment:
1 second seems sensible, I'm fairly sure the request goes to another container on the same machine.  I've never known it not respond immediately.